microservices-modules/micronaut/src/test/java/com/baeldung/micronaut/httpfilters/HelloFilterTest.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.micronaut.httpfilters;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import com.baeldung.micronaut.httpfilters.service.LogService;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.client.BlockingHttpClient;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.runtime.EmbeddedApplication;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+
+@MicronautTest
+class HelloFilterTest {
+
+    final String MESSAGE = "Hello message";
+
+    @Inject
+    EmbeddedApplication<?> application;
+
+    @Test
+    void testHelloMessageFilter(@Client("/") HttpClient httpClient) {
+        var appender = new MemoryAppender();
+        Logger l = (Logger) LoggerFactory.getLogger(LogService.class);
+        l.addAppender(appender);
+        appender.start();
```

Comment:
This  is not a typical way of verifying collaborators with Micronaut, they have a guide:
https://micronaut-projects.github.io/micronaut-test/latest/guide/#junit5MockingCollaborators

Showing how to use things like a Mockito mock via `@MockBean` and `@Inject`

You are mixing dependency injection techniques with a `private static final` SLF4J logger, which doesn't mix well and lead us to having workaround boilerplate that detracts from the main purpose of the test.

I recommend injected a `@Named` logger component in the `LogService`, to make this more directly testable using the standard Mockito verify technique